### PR TITLE
Merge CDM datasets into one

### PR DESCRIPTION
**Pull request by aahladmanas**
bq_utils.merge_tables now runs a copy job to merge multiple tables into a new one.
It's been tested with NYC and PITT 5 person data.

TODO:

1. Add polling/timeout parameters to the function
2. Test what happens when the tables to be merged don't have the same schema.

### CODE REVIEW

File: data_steward/bq_utils.py [hunk 1]
```diff
@@ -113,3 +118,74 @@ def get_job_details(job_id):
...
+    :source_dataset_name: dataset where all the source tables reside
+    :destination_table_name: data goes into this table
+    :destination_dataset_name: dataset where the destination table resides
+    :returns: True if successfull. Or False if error or taking too long.
+
+    """
+    app_id = app_identity.get_application_id()
+    job_body = { 
+        'configuration': {
+            "copy": { # "sourceTable": { "projectId": string, "datasetId": string, "tableId": string },
```
Comment (mark-velez) [inline data_steward/bq_utils.py]: I will remove commented out code

File: data_steward/bq_utils.py [hunk 2]
```diff
@@ -113,3 +118,74 @@ def get_job_details(job_id):
...
+                "sourceTables": [{
+                                        "projectId": app_id,
+                                        "datasetId": source_dataset_name,
+                                        "tableId": table_name
+                        } for table_name in source_table_name_list],
```
Comment (mark-velez) [inline data_steward/bq_utils.py]: I will make separate variable for readability

File: data_steward/bq_utils.py [hunk 3]
```diff
@@ -113,3 +118,74 @@ def get_job_details(job_id):
...
+            logging.info(' || '.join(error_messages))
+            return False,' || '.join(error_messages)
+    else:
+        logging.info("Wait timeout exceeded before load job with id '%s' was done" % job_id)
+        return False,"Job timeout"
+    return True,""
+
+
+def query_table(query_string, datasetId, tableId):
+
```
Comment (mark-velez) [inline data_steward/bq_utils.py]: This needs a docstring

Comment (mark-velez) [reply data_steward/bq_utils.py]: Please follow the conventions used in other areas

File: data_steward/test/unit_test/bq_utils_test.py [hunk 4]
```diff
@@ -42,7 +50,50 @@ def test_load_table_from_bucket_error_on_bad_table_name(self):
...
+        with open(PITT_FIVE_PERSONS_PERSON_CSV, 'rb') as fp:
+            gcs_utils.upload_object(gcs_utils.get_hpo_bucket('pitt'), 'person.csv', fp)
+        pitt_result = bq_utils.load_table_from_bucket('pitt', 'person')
+
+        nyc_person_ids =  [int(row['person_id']) for row in resources._csv_to_list(NYC_FIVE_PERSONS_PERSON_CSV)]
+        pitt_person_ids = [int(row['person_id']) for row in resources._csv_to_list(PITT_FIVE_PERSONS_PERSON_CSV)]
+        expected_result = nyc_person_ids + pitt_person_ids 
+        expected_result.sort()
+
+        time.sleep(5)
```
[the remark this answers is not in the document]
Comment (aahladmanas) [reply data_steward/test/unit_test/bq_utils_test.py]: This allows for the upload to complete.

File: data_steward/bq_utils.py (hunk 3 above)
Comment (aahladmanas) [reply data_steward/bq_utils.py]: What do you mean by other areas?

File: data_steward/bq_utils.py (hunk 1 above)
Comment (aahladmanas) [reply data_steward/bq_utils.py]: This is done.

File: data_steward/bq_utils.py (hunk 2 above)
Comment (aahladmanas) [reply data_steward/bq_utils.py]: Made this change.

**Comment by aahladmanas** [comment]
This ^ should read change as recommended in PR#14. Is this a good commit of fixes?

File: data_steward/bq_utils.py (hunk 3 above)
Comment (mark-velez) [reply data_steward/bq_utils.py]: Yes, my comment was unclear, sorry. I meant to follow how we document params in our other docstrings.

File: data_steward/test/unit_test/bq_utils_test.py [hunk 5]
```diff
@@ -4,7 +4,11 @@
 import bq_utils
 import gcs_utils
 from google.appengine.ext import testbed
-from test_util import FAKE_HPO_ID, FIVE_PERSONS_PERSON_CSV, NYC_FIVE_PERSONS_PERSON_CSV, PITT_FIVE_PERSONS_PERSON_CSV
+from test_util import FAKE_HPO_ID, \
```
Comment (mark-velez) [inline data_steward/test/unit_test/bq_utils_test.py]: Backslashes make me sad, I will reformat this to one-liner if that's ok with you...

Comment (aahladmanas) [reply data_steward/test/unit_test/bq_utils_test.py]: That's fine with me. Will follow that next time on.

**Pull request closed without merging**